tests/common/plugins/sanity_check/checks.py

```diff
@@ -278,6 +285,166 @@ def _check_monit_services_status(check_result, monit_services_status):
+
+    return intf_mac, mgmt_ipv4
+
+
+@pytest.fixture(scope='module')
+def check_mux_simulator(mux_server_url, ptf_server_intf, tor_mux_intf, ptfadapter, upper_tor_host, lower_tor_host):
+
+    def _check():
+        """
+        @summary: Checks if the OVS bridge mux simulator is functioning correctly
+        @return: A dictionary containing the testing result of the PTF interface tested:
+            {
+                'failed': <True/False>,
+                'failed_reason': <reason string>,
+                'intf': '<PTF interface name> mux simulator'
+            }
+        """
+        results = {
+                    'failed': False,
+                    'failed_reason': '',
+                    'check_item': '{} mux simulator'.format(ptf_server_intf)
+                }
+
+        logger.info("Checking mux simulator status for PTF interface {}".format(ptf_server_intf))
+        ptf_port_index = int(ptf_server_intf.replace('eth', ''))
+        mux_simulator_index = ptf_port_index + 1
+        recover_all_directions(mux_server_url, mux_simulator_index)
+
+        upper_tor_intf_mac, upper_tor_mgmt_ip = get_arp_pkt_info(upper_tor_host)
+        lower_tor_intf_mac, lower_tor_mgmt_ip = get_arp_pkt_info(lower_tor_host)
+
+        upper_tor_ping_tgt_ip = '10.10.10.1'
+        lower_tor_ping_tgt_ip = '10.10.10.2'
+        ptf_arp_tgt_ip = '10.10.10.3'
+        ping_cmd = 'ping -I {} {} -c 1 -W 1; true'
+
+        upper_tor_exp_pkt = testutils.simple_arp_packet(eth_dst='ff:ff:ff:ff:ff:ff',
+                                                        eth_src=upper_tor_intf_mac,
+                                                        ip_snd=upper_tor_mgmt_ip,
+                                                        ip_tgt=upper_tor_ping_tgt_ip,
+                                                        hw_snd=upper_tor_intf_mac)
+        lower_tor_exp_pkt = testutils.simple_arp_packet(eth_dst='ff:ff:ff:ff:ff:ff',
+                                                        eth_src=lower_tor_intf_mac,
+                                                        ip_snd=lower_tor_mgmt_ip,
+                                                        ip_tgt=lower_tor_ping_tgt_ip,
+                                                        hw_snd=lower_tor_intf_mac)
+        
+        ptf_arp_pkt = testutils.simple_arp_packet(ip_tgt=ptf_arp_tgt_ip,
+                                                ip_snd=ptf_arp_tgt_ip,
+                                                arp_op=2)
+
+        # Run tests with upper ToR active
+        toggle_simulator_port_to_upper_tor(mux_server_url, mux_simulator_index)
+
+        try:
+            pytest_assert(get_simulator_read_side(mux_server_url, mux_simulator_index) == 1)
+        except AssertionError:
+            results['failed'] = True
+            results['failed_reason'] = 'Unable to switch active link to upper ToR'
+            return results
+
+        # Ping from both ToRs, expect both messages to reach PTF
```

Comment:
`# Ping from both ToRs, expect both messages to reach PTF` -> `# Ping from both ToRs, expect only message from upper_tor to reach PTF`

Reply:
Ack, thanks!